lib/ui/dart_runtime_hooks.cc

```diff
@@ -199,6 +202,42 @@ void Logger_PrintString(Dart_NativeArguments args) {
   }
 }
 
+void SaveCompilationTrace(Dart_NativeArguments args) {
+  const char* filePath = NULL;
+  {
+    Dart_Handle result = Dart_StringToCString(
+        Dart_GetNativeArgument(args, 0), &filePath);
+    if (Dart_IsError(result)) {
+      Dart_PropagateError(result);
+      return;
+    }
+  }
+
+  uint8_t* buffer = nullptr;
+  intptr_t size = 0;
+  Dart_Handle result = Dart_SaveCompilationTrace(&buffer, &size);
+  if (Dart_IsError(result)) {
+    FXL_LOG(ERROR) << "Failed to save trace: " << Dart_GetError(result);
```

Comment:
Instead of logging on error, we propagate the error back to the caller in Dart code. The pattern typically followed is to return a string. In case of error, the string describes the error. In case of success, the return string is null. You can convert this scheme into idiomatic Dart code in `natives.dart`.

Reply:
Look at the `_futurize` call in Dart code to see existing uses of this pattern.

Reply:
Done. Although, would it be cleaner to return Dart Error object instead of converting them to string every time? Or perhaps even simply call Dart_ThrowException(error)?